src/main/java/grabber/sources/n78zaotl_com.java

```diff
@@ -0,0 +1,103 @@
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class n78zaotl_com implements Source {
+    private final String name = "78zaotl.com";
+    private final String url = "https://www.78zaotl.com/xk/";
+    private final boolean canHeadless = false;
+    private Novel novel;
+    private Document toc;
+
+    public n78zaotl_com(){}
+    public n78zaotl_com(Novel novel) {this.novel = novel;}
+
+    @Override
+    public String getName() {
+        return name;
+    }
+
+    @Override
+    public String getUrl() {
+        return url;
+    }
+
+    @Override
+    public boolean canHeadless() {
+        return canHeadless;
+    }
+
+    @Override
+    public List<Chapter> getChapterList() {
+        List<Chapter> chapterList = new ArrayList<>();
+        try {
+            toc = Jsoup.connect(novel.novelLink)
+                    .cookies(novel.cookies)
+                    .userAgent("Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0")
+                    .get();
+            Elements chapters = toc.select("div.row > div > p > a");
+            for (Element e : chapters) {
+                chapterList.add(new Chapter(e.text(), e.attr("abs:href")));
+            }
+        } catch (HttpStatusException httpEr) {
+            GrabberUtils.err(novel.window, GrabberUtils.getHTMLErrMsg(httpEr));
+        } catch (IOException e) {
+            GrabberUtils.err(novel.window, "Could not connect to webpage!", e);
+        } catch (NullPointerException e) {
+            GrabberUtils.err(novel.window, "Could not find expected selectors. Correct novel link?", e);
+        }
+        return chapterList;
+    }
+
+    @Override
+    public Element getChapterContent(Chapter chapter) {
+        Element body = null;
+        try {
+            Document doc = Jsoup.connect(chapter.chapterURL)
+                    .cookies(novel.cookies)
+                    .userAgent("Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:83.0) Gecko/20100101 Firefox/83.0")
+                    .get();
+            body = doc.select("div.row > div").first();
+        }catch (HttpStatusException httpEr) {
+            GrabberUtils.err(novel.window, GrabberUtils.getHTMLErrMsg(httpEr));
+        } catch (IOException e) {
+            GrabberUtils.err(novel.window, "Could not connect to webpage!", e);
+        }
+        return body;
+    }
+
+    @Override
+    public NovelMetadata getMetadata() {
+        NovelMetadata novelMetadata = new NovelMetadata();
+        if (toc != null) {
+            Element title = toc.selectFirst("div.container > h5");
+            //not possible to get description, everything is in one container div
+            //no cover image provided
+            //no author provided
+            novelMetadata.setTitle(title.text());
+            novelMetadata.setAuthor("");
+            novelMetadata.setDescription("");
```

Comment:
Synopsis is inside p tags

Reply:
> Synopsis is inside p tags

could i just select the p tags that come after the container, or do I have to merge them?
I committed what I think works, but let me know!